src/jit/codegenarm64.cpp

```diff
@@ -3438,42 +3452,45 @@ void CodeGen::genCodeForCompare(GenTreeOp* tree)
 
     genConsumeOperands(tree);
 
-    emitAttr cmpSize = EA_ATTR(genTypeSize(op1Type));
-
-    assert(genTypeSize(op1Type) == genTypeSize(op2Type));
-
-    if (varTypeIsFloating(op1Type))
+    if ((tree->gtFlags & GTF_USE_FLAGS) == 0)
     {
-        assert(varTypeIsFloating(op2Type));
```

Comment:
Rest of changes in the file are whitespace

Reply:
> It probably should be if ((op1-gtFlags & GTF_SET_FLAGS) == 0). That is, if op1 doesn't set flags we need to emit a compare instruction.

But isn't the point of the code in `Lowering` to set `GTF_USE_FLAGS` on the compare when it can simply use the flags set by the operand?

Reply:
OK